cmd/createtree/main.go

```diff
@@ -0,0 +1,188 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/golang/protobuf/ptypes"
+	"github.com/golang/protobuf/ptypes/any"
+	"github.com/google/trillian"
+	"github.com/google/trillian/crypto/sigpb"
+	"google.golang.org/grpc"
+)
+
+var (
+	adminEndpoint = flag.String("admin_endpoint", "", "Endpoint of the gRPC Trillian Admin Server")
+
+	treeState          = flag.String("tree_state", trillian.TreeState_ACTIVE.String(), "State of the new tree")
+	treeType           = flag.String("tree_type", trillian.TreeType_LOG.String(), "Type of the new tree")
+	hashStrategy       = flag.String("hash_strategy", trillian.HashStrategy_RFC_6962.String(), "Hash strategy (aka preimage protection) of the new tree")
+	hashAlgorithm      = flag.String("hash_algorithm", sigpb.DigitallySigned_SHA256.String(), "Hash algorithm of the new tree")
+	signatureAlgorithm = flag.String("signature_algorithm", sigpb.DigitallySigned_RSA.String(), "Signature algorithm of the new tree")
+	duplicatePolicy    = flag.String("duplicate_policy", trillian.DuplicatePolicy_DUPLICATES_NOT_ALLOWED.String(), "Duplicate policy of the new tree")
+	displayName        = flag.String("display_name", "", "Display name of the new tree")
+	description        = flag.String("description", "", "Description of the new tree")
+
+	privateKeyType = flag.String("private_key_type", "PEMKeyFile", "Type of private key to be used")
+	pemKeyPath     = flag.String("pem_key_path", "", "Path to the private key PEM file")
+	pemKeyPassword = flag.String("pem_key_password", "", "Password of the private key PEM file")
+)
+
+// println is used by tests to examine program output.
+var println = fmt.Println
+
+// createOpts contains all user-supplied options required to run the program.
+// It's meant to facilitate tests and focus flag reads to a single point.
+type createOpts struct {
+	endpoint                                                                                                  string
+	treeState, treeType, hashStrategy, hashAlgorithm, sigAlgorithm, duplicatePolicy, displayName, description string
+	privateKeyType, pemKeyPath, pemKeyPass                                                                    string
+}
+
+func createTree(opts *createOpts) (*trillian.Tree, error) {
+	req, err := newRequestFromFlags(opts)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := grpc.Dial(opts.endpoint, grpc.WithInsecure())
+	if err != nil {
+		return nil, err
+	}
+	defer conn.Close()
+
+	ctx := context.Background()
+	tree, err := trillian.NewTrillianAdminClient(conn).CreateTree(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	// DO NOT change the output format, scripts are meant to depend on it.
+	// If you really want to change it, provide an output_format flag and
+	// keep the default as-is.
+	println(tree.TreeId)
```

Comment:
I still think this should be separated out.  There are two steps:

 1) create a new Tree with a fresh ID
 2) given a tree, print information about it (specifically, its treeID)

Step 1 is more functional in style -- feed in inputs, get back tree/treeID.   

Step 2 involves I/O, and at the moment is sufficiently trivial that it doesn't need testing.  If we do have different output formats later, it will need testing -- but having it as a distinct step will make that testing easier (and independent).